lib/destroyed_at.rb

```diff
@@ -48,6 +49,12 @@ def destroy(timestamp = nil)
     end
   end
 
+  def mark_for_destruction(timestamp = nil)
+    @_pending_destroyed_at_timestamp = timestamp
```

Comment:
ActiveRecord creates an instance variable called `@marked_for_destruction` and this library uses `destroyed_at` as the timestamp property for destroyed objects. To more closely replicate how AR handles `mark_for_destruction` and the naming conventions in this library, can we rename `@_pending_destroyed_at_timestamp` to `@marked_for_destruction_at`?